packages/ui/src/RangeBar.tsx

```diff
@@ -0,0 +1,132 @@
+import React, {
+  useContext,
+  useState,
+  useEffect,
+  ForwardedRef,
+  forwardRef,
+  HTMLProps,
+} from 'react'
+import styled from 'styled-components'
+
+import { Text } from './Text'
+
+export interface RangeBarProps extends HTMLProps<HTMLInputElement> {
+  /**
+   * Step value of scroll bar
+   */
+  stepvalue?: number
+}
+
+/**
+ * Height of the scroll bar wrapper container
+ */
+const containerHeight = 70
+
+export const RangeBar = forwardRef(
+  (
+    {
+      type = 'range',
+      className = '',
+      disabled = false,
+      stepvalue,
+      min,
+      max,
+      ...props
+    }: RangeBarProps,
+    ref: ForwardedRef<HTMLInputElement>,
+  ) => {
+    const [valuePercent, setValuePercent] = useState(0)
+    const [rangeValue, setRangeValue] = useState(0)
+
+    const calculatePercent = (val: string) => {
+      setValuePercent((parseInt(val, 10) / Number(max)) * 100)
+    }
+
+    const handleRangeChange = e => {
+      if (e && e.target && e.target.value) {
+        setRangeValue(e.target.value)
+        calculatePercent(e.target.value)
+      }
+    }
+
+    useEffect(() => {
+      const startValue = (Number(max) - Number(min)) / 2
```

Comment:
startValue should be props.value (and set outside if it needs to have a startValue.